common4j/src/main/com/microsoft/identity/common/java/cache/SharedPreferencesAccountCredentialCacheWithMemoryCache.java

```diff
@@ -332,6 +339,10 @@ public boolean test(String value) {
                 // Remove the entry and return null...
                 Logger.warn(methodTag, "The returned Credential was uninitialized. Removing...");
                 mSharedPreferencesFileManager.remove(cacheKey);
+            } else if (AccessTokenRecord.class == clazz
+                    && !mSha1Cleared
+                    && isSha1ApplicationIdentifierAccessToken(credential)) {
+                    mSharedPreferencesFileManager.remove(cacheKey);
```

Comment:
>mSharedPreferencesFileManager.remove(cacheKey); [](http://example.com/codeflow?start=20&length=47)

I don't remember the iterator rules for safety in java.  Is it safe to remove an item from a collection while iterating over it -- or does it need to be a separate step?

Reply:
I guess we're doing it a few lines above so it can't be that unsafe :) Consider it closed.

Reply:
Looking at how `mSharedPreferencesFileManager.getAllFilteredByKey` is implemented, the iterator does refer to a Map that's directly returned from shared preferences, so I'm not sure either if this is actually safe. Though it doesn't really apply to this particular PR anymore because I removed the referenced line. 
I suppose if the existing remove() call hasn't caused any issues so far, seems fine for now?